recipe_scrapers/recipeland.py

```diff
@@ -0,0 +1,55 @@
+from ._abstract import AbstractScraper
+from ._grouping_utils import IngredientGroup
+from ._utils import normalize_string
+
+
+class RecipeLand(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "recipeland.com"
+
+    def _parse_ingredient_row(self, row):
+        amount = row.select_one(".amount") and row.select_one(".amount").get_text()
+        measure = row.select_one(".measure") and row.select_one(".measure").get_text()
+        ingredient_elem = row.select_one(".ingred")
+        ingredient = ingredient_elem and ingredient_elem.get_text()
+
+        full_ingredient = f"{amount} {measure} {ingredient}"
+        return normalize_string(full_ingredient)
+
+    def ingredients(self):
+        ingredient_rows = self.soup.select("#ingredient_list tbody tr")
+        return [
+            self._parse_ingredient_row(row)
+            for row in ingredient_rows
+            if not row.select_one("th.i-head")
+        ]
+
+    def ingredient_groups(self):
+
+        def add_group():
```

Comment:
This add_group() inner function relies on variables from the outer scope. The mutation of outer scope variables makes the flow harder to follow. Also it's called in two places which suggests the control flow could be simplified.

I know I'm a bit vague but can you try to refactor this a bit? Consider refactoring `add_group()` to take explicit parameters rather than using closure variables being my main point of concern.